torch/distributed/_composable/fsdp/fully_shard.py

```diff
@@ -83,3 +138,13 @@ def __new__(cls, *args, **kwargs):
         self = orig_cls.__new__(orig_cls, *args, **kwargs)
         self.__init__(*args, **kwargs)
         return self
+
+    def reshard(self) -> None:
+        """
+        Reshards the module's parameters by freeing unsharded parameters if
```

Comment:
if needed here means "the root state" won't be freed even if user call this API?

Reply:
If the user calls this on the root state, the root state's parameters _will_ be freed. This is to give users a way to actually do that if they want to despite the default behavior to not reshard after forward for the root.